distributed/deploy/cluster.py

```diff
@@ -449,3 +456,9 @@ def requested(self):
     @property
     def observed(self):
         return {d["name"] for d in self.scheduler_info["workers"].values()}
+
+    def __eq__(self, other):
+        return self.name == other.name
+
+    def __hash__(self):
+        return id(self)
```

Comment:
This caused me to think for a bit, but I convinced myself that it was probably ok.